pkg/cli/admin/top/persistentvolumeclaims.go

```diff
@@ -0,0 +1,342 @@
+}
+
+func (o *options) Run(ctx context.Context, args []string) error {
+	persistentVolumeClaimsBytes, err := GetPersistentVolumeClaims(ctx, o.getRoute, o.RESTConfig.BearerToken, o.Namespace, o.InsecureTLS, args)
+	if err != nil {
+		return err
+	}
+	promOutput := &PromOutput{}
+	err = json.Unmarshal([]byte(persistentVolumeClaimsBytes), &promOutput)
+	if err != nil {
+		return err
+	}
+	if len(promOutput.Data.Result) == 0 {
+		if o.Namespace == "" {
+			return fmt.Errorf("no persistentvolumeclaims found.")
+		} else {
+			if len(args) == 0 {
+				return fmt.Errorf("no persistentvolumeclaims found in %s namespace.", o.Namespace)
+			} else {
+				return fmt.Errorf("persistentvolumeclaim %q not found in %s namespace.", args[0], o.Namespace)
+			}
+		}
+	}
+
+	// if more pvc are requested as args but one of them does not not exist
+	if len(args) != 0 && len(promOutput.Data.Result) != len(args) {
+		resultingPvc := make(map[string]bool)
+		for _, _promOutputDataResult := range promOutput.Data.Result {
+			pvcName, _ := _promOutputDataResult.Metric["persistentvolumeclaim"]
+			resultingPvc[pvcName] = true
+		}
+		for _, arg := range args {
+			_, pvcPresentInResult := resultingPvc[arg]
+			if !pvcPresentInResult {
+				return fmt.Errorf("persistentvolumeclaim %q not found in %s namespace.", arg, o.Namespace)
+			}
+		}
+
+	}
+	headers := []string{"NAMESPACE", "NAME", "USAGE(%)"}
+	infos := []Info{}
+	for _, _promOutputDataResult := range promOutput.Data.Result {
+		namespaceName, _ := _promOutputDataResult.Metric["namespace"]
+		pvcName, _ := _promOutputDataResult.Metric["persistentvolumeclaim"]
+		usagePercentage := _promOutputDataResult.Value[1]
+		valueFloatLong, _ := strconv.ParseFloat(usagePercentage.(string), 64)
+		valueFloat := fmt.Sprintf("%.2f", valueFloatLong)
+		infos = append(infos, persistentVolumeClaimInfo{Namespace: namespaceName, Name: pvcName, UsagePercentage: valueFloat})
+
+	}
+	Print(o.Out, headers, infos)
+	return nil
+}
+
+func GetPersistentVolumeClaims(ctx context.Context, getRoute RouteGetter, bearerToken string, namespace string, insecureTLS bool, args []string) ([]byte, error) {
+	uri := &url.URL{
+		Scheme: "https",
+		Path:   "/api/v1/query",
+	}
+	query := ""
+	claimNames := ".*"
+	if namespace != "" {
+		if len(args) > 0 {
+			claimNames = strings.Join(args, "|")
+		}
+		query = fmt.Sprintf(`100*kubelet_volume_stats_used_bytes{persistentvolumeclaim=~"%s", namespace="%s"}/kubelet_volume_stats_capacity_bytes{persistentvolumeclaim=~"%s", namespace="%s"}`, claimNames, namespace, claimNames, namespace)
+	} else {
+		query = `100*kubelet_volume_stats_used_bytes{persistentvolumeclaim=~".*"}/kubelet_volume_stats_capacity_bytes{persistentvolumeclaim=~".*"}`
+	}
+	urlParams := url.Values{}
+	urlParams.Set("query", query)
+	uri.RawQuery = urlParams.Encode()
+
+	persistentVolumeClaimsBytes, err := getWithBearer(ctx, getRoute, "openshift-monitoring", "prometheus-k8s", uri, bearerToken, insecureTLS)
```

Comment:
AFAIU the [metricsclientset](https://github.com/kubernetes/metrics) API allows consumers to access only [`resource metrics (CPU and memory) for pods and nodes.`](https://github.com/kubernetes/metrics?tab=readme-ov-file#resource-metrics-api)

So if we strictly want `system:admin` users authenticated using the `KUBECONFIG` file to use that command we should evaluate:
- getting a token from a different place, maybe a secret containing it?! (I will try to check if any exists)
- using a different workflow to obtain this information rather than Prometheus API.

Reply:
I found out that the token from the secret `localhost-recovery-client-token` in `openshift-kube-apiserver` namespace is a valid one to use:
```
oc get secret -n openshift-kube-apiserver localhost-recovery-client-token -o json | jq '.data.token' -r | base64 -d
```

Reply:
I'll make this modification before next week so we can try to merge this PR, as @tsmetana agreed.

Reply:
Done!

https://github.com/openshift/oc/blob/a3cf81477f08ddbe8be66a49b620e94fc0c71823/pkg/cli/admin/top/persistentvolumeclaims.go#L150-L165

Reply:
Thanks!